lib/auth/apiserver.go

```diff
@@ -571,22 +571,25 @@ func (s *APIServer) generateKeyPair(auth ClientI, w http.ResponseWriter, r *http
 }
 
 type generateHostCertReq struct {
-	Key        []byte         `json:"key"`
-	Hostname   string         `json:"hostname"`
-	AuthDomain string         `json:"auth_domain"`
-	Roles      teleport.Roles `json:"roles"`
-	TTL        time.Duration  `json:"ttl"`
+	Key         []byte         `json:"key"`
+	HostID      string         `json:"host_id"`
+	NodeName    string         `json:"node_name"`
+	ClusterName string         `json:"cluster_name"`
```

Comment:
I would change the data structure name, but keep the json version of it to `auth_domain` or write the migration procedure.